packages/synapse-react-client/src/lib/containers/auth/TOTPForm.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { Box, Button } from '@mui/material'
+import { MuiOtpInput } from 'mui-one-time-password-input'
+
+const TOTP_LENGTH = 6
+const DIGIT_CHARACTERS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
+
+type TOTPFormProps = {
+  onSubmit: (value: string) => void
+}
+
+export default function TOTPForm(props: TOTPFormProps) {
+  const { onSubmit } = props
+  const [verificationCode, setVerificationCode] = React.useState('')
+  return (
+    <Box>
+      <MuiOtpInput
```

Comment:
I tried to roll my own OTP component but using this library ended up being 100x easier